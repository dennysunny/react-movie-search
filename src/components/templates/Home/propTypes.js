import {
  array,
  bool,
  object,
  oneOfType,
} from 'prop-types';

export const propTypes = {
  popular: array.isRequired,
  nowPlayingLoading: bool,
  popularLoading: bool,
  upcomingLoading: bool,
  nowplaying: oneOfType([object, array]),
  upcoming: oneOfType([object, array]),
};

export const defaultProps = {
  nowplaying: [],
  upcoming: [],
};
