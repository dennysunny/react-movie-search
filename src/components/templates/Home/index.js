import React, { Component } from 'react';

import { propTypes, defaultProps } from './propTypes';
import Spinner from '../../atoms/Spinner';
import PageTransition from '../../atoms/PageTransition';
import TopRatedMovies from '../../organisms/TopRatedMovies';
import HomeList from '../../atoms/HomeList';

export default class Home extends Component {
  static defaultProps = defaultProps;

  static propTypes = propTypes;

  render() {
    const {
      nowPlayingLoading,
      popularLoading,
      upcomingLoading,
      nowplaying,
      upcoming,
      popular,
    } = this.props;
    if (nowPlayingLoading || popularLoading || upcomingLoading) {
      return <Spinner />;
    }

    return (
      <PageTransition>
        <TopRatedMovies
          popular={popular}
        />

        <div className="top-lists-wrapper">
          <HomeList
            list={nowplaying.slice(0, 10)}
            title="In Theatres"
          />
          <HomeList
            list={upcoming.slice(0, 10)}
            title="Upcoming Movies"
          />
        </div>
      </PageTransition>
    );
  }
}
