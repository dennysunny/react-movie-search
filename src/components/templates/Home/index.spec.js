import React from 'react';
import { shallow } from 'enzyme';

import Home from '.';

const mockProps = {
  popular: [{
    id: 1,
    backdrop_path: 'path/to/image',
    poster_path: 'path/to/image',
    title: 'title',
    overview: 'overview',
    release_date: '01.01.2010',
    vote_average: 1,
    genre_ids: [1],
  },
  {
    id: 1,
    backdrop_path: 'path/to/image',
    poster_path: 'path/to/image',
    title: 'title',
    overview: 'overview',
    release_date: '01.01.2010',
    vote_average: 1,
    genre_ids: [1],
  },
  {
    id: 1,
    backdrop_path: 'path/to/image',
    poster_path: 'path/to/image',
    title: 'title',
    overview: 'overview',
    release_date: '01.01.2010',
    vote_average: 1,
    genre_ids: [1],
  },
  {
    id: 1,
    backdrop_path: 'path/to/image',
    poster_path: 'path/to/image',
    title: 'title',
    overview: 'overview',
    release_date: '01.01.2010',
    vote_average: 1,
    genre_ids: [1],
  },
  {
    id: 1,
    backdrop_path: 'path/to/image',
    poster_path: 'path/to/image',
    title: 'title',
    overview: 'overview',
    release_date: '01.01.2010',
    vote_average: 1,
    genre_ids: [1],
  },
  {
    id: 1,
    backdrop_path: 'path/to/image',
    poster_path: 'path/to/image',
    title: 'title',
    overview: 'overview',
    release_date: '01.01.2010',
    vote_average: 1,
    genre_ids: [1],
  },
  {
    id: 1,
    backdrop_path: 'path/to/image',
    poster_path: 'path/to/image',
    title: 'title',
    overview: 'overview',
    release_date: '01.01.2010',
    vote_average: 1,
    genre_ids: [1],
  },
  {
    id: 1,
    backdrop_path: 'path/to/image',
    poster_path: 'path/to/image',
    title: 'title',
    overview: 'overview',
    release_date: '01.01.2010',
    vote_average: 1,
    genre_ids: [2],
  }],
  popularLoading: false,
  nowPlayingLoading: false,
  upcomingLoading: false,
};

describe('Templates/Home', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<Home {...mockProps} />);
    wrapper.setProps({ ...mockProps });
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle filter function', () => {
    const wrapper = shallow(<Home {...mockProps} />);
    wrapper.setProps({ ...mockProps });
    wrapper.instance().filterTopMovies(2);

    expect(wrapper.state().topmovies.length).toEqual(1);
  });

  it('should handle filter function for more than 5 result', () => {
    const wrapper = shallow(<Home {...mockProps} />);
    wrapper.setProps({ ...mockProps });
    wrapper.instance().filterTopMovies(1);

    expect(wrapper.state().topmovies.length).toEqual(5);
  });

  it('should render spinner when loading', () => {
    const props = {
      nowPlayingLoading: true,
    };
    const wrapper = shallow(<Home {...mockProps} {...props} />);
    wrapper.setProps({ ...mockProps });
    expect(wrapper).toMatchSnapshot();
  });
});
