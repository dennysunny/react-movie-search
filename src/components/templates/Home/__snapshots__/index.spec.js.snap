// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates/Home should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    nowPlayingLoading={false}
    nowplaying={Array []}
    popular={
      Array [
        Object {
          "backdrop_path": "path/to/image",
          "genre_ids": Array [
            1,
          ],
          "id": 1,
          "overview": "overview",
          "poster_path": "path/to/image",
          "release_date": "01.01.2010",
          "title": "title",
          "vote_average": 1,
        },
        Object {
          "backdrop_path": "path/to/image",
          "genre_ids": Array [
            1,
          ],
          "id": 1,
          "overview": "overview",
          "poster_path": "path/to/image",
          "release_date": "01.01.2010",
          "title": "title",
          "vote_average": 1,
        },
        Object {
          "backdrop_path": "path/to/image",
          "genre_ids": Array [
            1,
          ],
          "id": 1,
          "overview": "overview",
          "poster_path": "path/to/image",
          "release_date": "01.01.2010",
          "title": "title",
          "vote_average": 1,
        },
        Object {
          "backdrop_path": "path/to/image",
          "genre_ids": Array [
            1,
          ],
          "id": 1,
          "overview": "overview",
          "poster_path": "path/to/image",
          "release_date": "01.01.2010",
          "title": "title",
          "vote_average": 1,
        },
        Object {
          "backdrop_path": "path/to/image",
          "genre_ids": Array [
            1,
          ],
          "id": 1,
          "overview": "overview",
          "poster_path": "path/to/image",
          "release_date": "01.01.2010",
          "title": "title",
          "vote_average": 1,
        },
        Object {
          "backdrop_path": "path/to/image",
          "genre_ids": Array [
            1,
          ],
          "id": 1,
          "overview": "overview",
          "poster_path": "path/to/image",
          "release_date": "01.01.2010",
          "title": "title",
          "vote_average": 1,
        },
        Object {
          "backdrop_path": "path/to/image",
          "genre_ids": Array [
            1,
          ],
          "id": 1,
          "overview": "overview",
          "poster_path": "path/to/image",
          "release_date": "01.01.2010",
          "title": "title",
          "vote_average": 1,
        },
        Object {
          "backdrop_path": "path/to/image",
          "genre_ids": Array [
            2,
          ],
          "id": 1,
          "overview": "overview",
          "poster_path": "path/to/image",
          "release_date": "01.01.2010",
          "title": "title",
          "vote_average": 1,
        },
      ]
    }
    popularLoading={false}
    upcoming={Array []}
    upcomingLoading={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <TopRatedMovies
          filterTopMovies={[Function]}
          topMovies={
            Array [
              Object {
                "backdrop_path": "path/to/image",
                "genre_ids": Array [
                  1,
                ],
                "id": 1,
                "overview": "overview",
                "poster_path": "path/to/image",
                "release_date": "01.01.2010",
                "title": "title",
                "vote_average": 1,
              },
              Object {
                "backdrop_path": "path/to/image",
                "genre_ids": Array [
                  1,
                ],
                "id": 1,
                "overview": "overview",
                "poster_path": "path/to/image",
                "release_date": "01.01.2010",
                "title": "title",
                "vote_average": 1,
              },
              Object {
                "backdrop_path": "path/to/image",
                "genre_ids": Array [
                  1,
                ],
                "id": 1,
                "overview": "overview",
                "poster_path": "path/to/image",
                "release_date": "01.01.2010",
                "title": "title",
                "vote_average": 1,
              },
              Object {
                "backdrop_path": "path/to/image",
                "genre_ids": Array [
                  1,
                ],
                "id": 1,
                "overview": "overview",
                "poster_path": "path/to/image",
                "release_date": "01.01.2010",
                "title": "title",
                "vote_average": 1,
              },
              Object {
                "backdrop_path": "path/to/image",
                "genre_ids": Array [
                  1,
                ],
                "id": 1,
                "overview": "overview",
                "poster_path": "path/to/image",
                "release_date": "01.01.2010",
                "title": "title",
                "vote_average": 1,
              },
            ]
          }
        />,
        <div
          className="top-lists-wrapper"
        >
          <HomeList
            list={Array []}
            title="In Theatres"
          />
          <HomeList
            list={Array []}
            title="Upcoming Movies"
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "filterTopMovies": [Function],
          "topMovies": Array [
            Object {
              "backdrop_path": "path/to/image",
              "genre_ids": Array [
                1,
              ],
              "id": 1,
              "overview": "overview",
              "poster_path": "path/to/image",
              "release_date": "01.01.2010",
              "title": "title",
              "vote_average": 1,
            },
            Object {
              "backdrop_path": "path/to/image",
              "genre_ids": Array [
                1,
              ],
              "id": 1,
              "overview": "overview",
              "poster_path": "path/to/image",
              "release_date": "01.01.2010",
              "title": "title",
              "vote_average": 1,
            },
            Object {
              "backdrop_path": "path/to/image",
              "genre_ids": Array [
                1,
              ],
              "id": 1,
              "overview": "overview",
              "poster_path": "path/to/image",
              "release_date": "01.01.2010",
              "title": "title",
              "vote_average": 1,
            },
            Object {
              "backdrop_path": "path/to/image",
              "genre_ids": Array [
                1,
              ],
              "id": 1,
              "overview": "overview",
              "poster_path": "path/to/image",
              "release_date": "01.01.2010",
              "title": "title",
              "vote_average": 1,
            },
            Object {
              "backdrop_path": "path/to/image",
              "genre_ids": Array [
                1,
              ],
              "id": 1,
              "overview": "overview",
              "poster_path": "path/to/image",
              "release_date": "01.01.2010",
              "title": "title",
              "vote_average": 1,
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <HomeList
              list={Array []}
              title="In Theatres"
            />,
            <HomeList
              list={Array []}
              title="Upcoming Movies"
            />,
          ],
          "className": "top-lists-wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "list": Array [],
              "title": "In Theatres",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "list": Array [],
              "title": "Upcoming Movies",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <TopRatedMovies
            filterTopMovies={[Function]}
            topMovies={
              Array [
                Object {
                  "backdrop_path": "path/to/image",
                  "genre_ids": Array [
                    1,
                  ],
                  "id": 1,
                  "overview": "overview",
                  "poster_path": "path/to/image",
                  "release_date": "01.01.2010",
                  "title": "title",
                  "vote_average": 1,
                },
                Object {
                  "backdrop_path": "path/to/image",
                  "genre_ids": Array [
                    1,
                  ],
                  "id": 1,
                  "overview": "overview",
                  "poster_path": "path/to/image",
                  "release_date": "01.01.2010",
                  "title": "title",
                  "vote_average": 1,
                },
                Object {
                  "backdrop_path": "path/to/image",
                  "genre_ids": Array [
                    1,
                  ],
                  "id": 1,
                  "overview": "overview",
                  "poster_path": "path/to/image",
                  "release_date": "01.01.2010",
                  "title": "title",
                  "vote_average": 1,
                },
                Object {
                  "backdrop_path": "path/to/image",
                  "genre_ids": Array [
                    1,
                  ],
                  "id": 1,
                  "overview": "overview",
                  "poster_path": "path/to/image",
                  "release_date": "01.01.2010",
                  "title": "title",
                  "vote_average": 1,
                },
                Object {
                  "backdrop_path": "path/to/image",
                  "genre_ids": Array [
                    1,
                  ],
                  "id": 1,
                  "overview": "overview",
                  "poster_path": "path/to/image",
                  "release_date": "01.01.2010",
                  "title": "title",
                  "vote_average": 1,
                },
              ]
            }
          />,
          <div
            className="top-lists-wrapper"
          >
            <HomeList
              list={Array []}
              title="In Theatres"
            />
            <HomeList
              list={Array []}
              title="Upcoming Movies"
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "filterTopMovies": [Function],
            "topMovies": Array [
              Object {
                "backdrop_path": "path/to/image",
                "genre_ids": Array [
                  1,
                ],
                "id": 1,
                "overview": "overview",
                "poster_path": "path/to/image",
                "release_date": "01.01.2010",
                "title": "title",
                "vote_average": 1,
              },
              Object {
                "backdrop_path": "path/to/image",
                "genre_ids": Array [
                  1,
                ],
                "id": 1,
                "overview": "overview",
                "poster_path": "path/to/image",
                "release_date": "01.01.2010",
                "title": "title",
                "vote_average": 1,
              },
              Object {
                "backdrop_path": "path/to/image",
                "genre_ids": Array [
                  1,
                ],
                "id": 1,
                "overview": "overview",
                "poster_path": "path/to/image",
                "release_date": "01.01.2010",
                "title": "title",
                "vote_average": 1,
              },
              Object {
                "backdrop_path": "path/to/image",
                "genre_ids": Array [
                  1,
                ],
                "id": 1,
                "overview": "overview",
                "poster_path": "path/to/image",
                "release_date": "01.01.2010",
                "title": "title",
                "vote_average": 1,
              },
              Object {
                "backdrop_path": "path/to/image",
                "genre_ids": Array [
                  1,
                ],
                "id": 1,
                "overview": "overview",
                "poster_path": "path/to/image",
                "release_date": "01.01.2010",
                "title": "title",
                "vote_average": 1,
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <HomeList
                list={Array []}
                title="In Theatres"
              />,
              <HomeList
                list={Array []}
                title="Upcoming Movies"
              />,
            ],
            "className": "top-lists-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "list": Array [],
                "title": "In Theatres",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "list": Array [],
                "title": "Upcoming Movies",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Templates/Home should render spinner when loading 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home
    nowPlayingLoading={false}
    nowplaying={Array []}
    popular={
      Array [
        Object {
          "backdrop_path": "path/to/image",
          "genre_ids": Array [
            1,
          ],
          "id": 1,
          "overview": "overview",
          "poster_path": "path/to/image",
          "release_date": "01.01.2010",
          "title": "title",
          "vote_average": 1,
        },
        Object {
          "backdrop_path": "path/to/image",
          "genre_ids": Array [
            1,
          ],
          "id": 1,
          "overview": "overview",
          "poster_path": "path/to/image",
          "release_date": "01.01.2010",
          "title": "title",
          "vote_average": 1,
        },
        Object {
          "backdrop_path": "path/to/image",
          "genre_ids": Array [
            1,
          ],
          "id": 1,
          "overview": "overview",
          "poster_path": "path/to/image",
          "release_date": "01.01.2010",
          "title": "title",
          "vote_average": 1,
        },
        Object {
          "backdrop_path": "path/to/image",
          "genre_ids": Array [
            1,
          ],
          "id": 1,
          "overview": "overview",
          "poster_path": "path/to/image",
          "release_date": "01.01.2010",
          "title": "title",
          "vote_average": 1,
        },
        Object {
          "backdrop_path": "path/to/image",
          "genre_ids": Array [
            1,
          ],
          "id": 1,
          "overview": "overview",
          "poster_path": "path/to/image",
          "release_date": "01.01.2010",
          "title": "title",
          "vote_average": 1,
        },
        Object {
          "backdrop_path": "path/to/image",
          "genre_ids": Array [
            1,
          ],
          "id": 1,
          "overview": "overview",
          "poster_path": "path/to/image",
          "release_date": "01.01.2010",
          "title": "title",
          "vote_average": 1,
        },
        Object {
          "backdrop_path": "path/to/image",
          "genre_ids": Array [
            1,
          ],
          "id": 1,
          "overview": "overview",
          "poster_path": "path/to/image",
          "release_date": "01.01.2010",
          "title": "title",
          "vote_average": 1,
        },
        Object {
          "backdrop_path": "path/to/image",
          "genre_ids": Array [
            2,
          ],
          "id": 1,
          "overview": "overview",
          "poster_path": "path/to/image",
          "release_date": "01.01.2010",
          "title": "title",
          "vote_average": 1,
        },
      ]
    }
    popularLoading={false}
    upcoming={Array []}
    upcomingLoading={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <TopRatedMovies
          filterTopMovies={[Function]}
          topMovies={
            Array [
              Object {
                "backdrop_path": "path/to/image",
                "genre_ids": Array [
                  1,
                ],
                "id": 1,
                "overview": "overview",
                "poster_path": "path/to/image",
                "release_date": "01.01.2010",
                "title": "title",
                "vote_average": 1,
              },
              Object {
                "backdrop_path": "path/to/image",
                "genre_ids": Array [
                  1,
                ],
                "id": 1,
                "overview": "overview",
                "poster_path": "path/to/image",
                "release_date": "01.01.2010",
                "title": "title",
                "vote_average": 1,
              },
              Object {
                "backdrop_path": "path/to/image",
                "genre_ids": Array [
                  1,
                ],
                "id": 1,
                "overview": "overview",
                "poster_path": "path/to/image",
                "release_date": "01.01.2010",
                "title": "title",
                "vote_average": 1,
              },
              Object {
                "backdrop_path": "path/to/image",
                "genre_ids": Array [
                  1,
                ],
                "id": 1,
                "overview": "overview",
                "poster_path": "path/to/image",
                "release_date": "01.01.2010",
                "title": "title",
                "vote_average": 1,
              },
              Object {
                "backdrop_path": "path/to/image",
                "genre_ids": Array [
                  1,
                ],
                "id": 1,
                "overview": "overview",
                "poster_path": "path/to/image",
                "release_date": "01.01.2010",
                "title": "title",
                "vote_average": 1,
              },
            ]
          }
        />,
        <div
          className="top-lists-wrapper"
        >
          <HomeList
            list={Array []}
            title="In Theatres"
          />
          <HomeList
            list={Array []}
            title="Upcoming Movies"
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "filterTopMovies": [Function],
          "topMovies": Array [
            Object {
              "backdrop_path": "path/to/image",
              "genre_ids": Array [
                1,
              ],
              "id": 1,
              "overview": "overview",
              "poster_path": "path/to/image",
              "release_date": "01.01.2010",
              "title": "title",
              "vote_average": 1,
            },
            Object {
              "backdrop_path": "path/to/image",
              "genre_ids": Array [
                1,
              ],
              "id": 1,
              "overview": "overview",
              "poster_path": "path/to/image",
              "release_date": "01.01.2010",
              "title": "title",
              "vote_average": 1,
            },
            Object {
              "backdrop_path": "path/to/image",
              "genre_ids": Array [
                1,
              ],
              "id": 1,
              "overview": "overview",
              "poster_path": "path/to/image",
              "release_date": "01.01.2010",
              "title": "title",
              "vote_average": 1,
            },
            Object {
              "backdrop_path": "path/to/image",
              "genre_ids": Array [
                1,
              ],
              "id": 1,
              "overview": "overview",
              "poster_path": "path/to/image",
              "release_date": "01.01.2010",
              "title": "title",
              "vote_average": 1,
            },
            Object {
              "backdrop_path": "path/to/image",
              "genre_ids": Array [
                1,
              ],
              "id": 1,
              "overview": "overview",
              "poster_path": "path/to/image",
              "release_date": "01.01.2010",
              "title": "title",
              "vote_average": 1,
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <HomeList
              list={Array []}
              title="In Theatres"
            />,
            <HomeList
              list={Array []}
              title="Upcoming Movies"
            />,
          ],
          "className": "top-lists-wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "list": Array [],
              "title": "In Theatres",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "list": Array [],
              "title": "Upcoming Movies",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <TopRatedMovies
            filterTopMovies={[Function]}
            topMovies={
              Array [
                Object {
                  "backdrop_path": "path/to/image",
                  "genre_ids": Array [
                    1,
                  ],
                  "id": 1,
                  "overview": "overview",
                  "poster_path": "path/to/image",
                  "release_date": "01.01.2010",
                  "title": "title",
                  "vote_average": 1,
                },
                Object {
                  "backdrop_path": "path/to/image",
                  "genre_ids": Array [
                    1,
                  ],
                  "id": 1,
                  "overview": "overview",
                  "poster_path": "path/to/image",
                  "release_date": "01.01.2010",
                  "title": "title",
                  "vote_average": 1,
                },
                Object {
                  "backdrop_path": "path/to/image",
                  "genre_ids": Array [
                    1,
                  ],
                  "id": 1,
                  "overview": "overview",
                  "poster_path": "path/to/image",
                  "release_date": "01.01.2010",
                  "title": "title",
                  "vote_average": 1,
                },
                Object {
                  "backdrop_path": "path/to/image",
                  "genre_ids": Array [
                    1,
                  ],
                  "id": 1,
                  "overview": "overview",
                  "poster_path": "path/to/image",
                  "release_date": "01.01.2010",
                  "title": "title",
                  "vote_average": 1,
                },
                Object {
                  "backdrop_path": "path/to/image",
                  "genre_ids": Array [
                    1,
                  ],
                  "id": 1,
                  "overview": "overview",
                  "poster_path": "path/to/image",
                  "release_date": "01.01.2010",
                  "title": "title",
                  "vote_average": 1,
                },
              ]
            }
          />,
          <div
            className="top-lists-wrapper"
          >
            <HomeList
              list={Array []}
              title="In Theatres"
            />
            <HomeList
              list={Array []}
              title="Upcoming Movies"
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "filterTopMovies": [Function],
            "topMovies": Array [
              Object {
                "backdrop_path": "path/to/image",
                "genre_ids": Array [
                  1,
                ],
                "id": 1,
                "overview": "overview",
                "poster_path": "path/to/image",
                "release_date": "01.01.2010",
                "title": "title",
                "vote_average": 1,
              },
              Object {
                "backdrop_path": "path/to/image",
                "genre_ids": Array [
                  1,
                ],
                "id": 1,
                "overview": "overview",
                "poster_path": "path/to/image",
                "release_date": "01.01.2010",
                "title": "title",
                "vote_average": 1,
              },
              Object {
                "backdrop_path": "path/to/image",
                "genre_ids": Array [
                  1,
                ],
                "id": 1,
                "overview": "overview",
                "poster_path": "path/to/image",
                "release_date": "01.01.2010",
                "title": "title",
                "vote_average": 1,
              },
              Object {
                "backdrop_path": "path/to/image",
                "genre_ids": Array [
                  1,
                ],
                "id": 1,
                "overview": "overview",
                "poster_path": "path/to/image",
                "release_date": "01.01.2010",
                "title": "title",
                "vote_average": 1,
              },
              Object {
                "backdrop_path": "path/to/image",
                "genre_ids": Array [
                  1,
                ],
                "id": 1,
                "overview": "overview",
                "poster_path": "path/to/image",
                "release_date": "01.01.2010",
                "title": "title",
                "vote_average": 1,
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <HomeList
                list={Array []}
                title="In Theatres"
              />,
              <HomeList
                list={Array []}
                title="Upcoming Movies"
              />,
            ],
            "className": "top-lists-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "list": Array [],
                "title": "In Theatres",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "list": Array [],
                "title": "Upcoming Movies",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
